File: TestUDP/Assets/Scripts/Data/ClientMessage.cs
public enum ClientMessageType
{
    authenticate, position
}

public abstract class ClientMessage
{
    public string id = NetworkManager.current.GetId;
    public ClientMessageType type;

    protected ClientMessage(ClientMessageType type) {
        this.type = type;
    }
}

public class ClientMessagePosition : ClientMessage
{
    public Vector3Data position;

    public ClientMessagePosition(Vector3Data position) : base(ClientMessageType.position) {
        this.position = position;
    }
}

public class ClientMessageAuthenticate : ClientMessage
{
    public ClientMessageAuthenticate() : base(ClientMessageType.authenticate) { }
}

File: API/data/client_message.ts
import { Vector3 } from "./vector3"

enum ClientMessageType { authenticate, position }

abstract class ClientMessage {
    readonly id: string
    readonly type: ClientMessageType
}

class ClientMessagePosition extends ClientMessage {
    readonly position: Vector3
}

class ClientMessageAuthenticate extends ClientMessage { }

export { ClientMessageType, ClientMessage, ClientMessagePosition, ClientMessageAuthenticate }